src/test/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesTaskLauncherIntegrationIT.java

```diff
@@ -16,235 +16,122 @@
 
 package org.springframework.cloud.deployer.spi.kubernetes;
 
-import java.time.Duration;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
-import java.util.Optional;
-import java.util.UUID;
+import java.util.function.Consumer;
 
 import io.fabric8.kubernetes.api.model.Container;
 import io.fabric8.kubernetes.api.model.Pod;
-import io.fabric8.kubernetes.client.KubernetesClient;
-import org.junit.jupiter.api.Disabled;
+import io.fabric8.kubernetes.client.utils.PodStatusUtil;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
 
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.cloud.deployer.resource.docker.DockerResource;
 import org.springframework.cloud.deployer.spi.core.AppDefinition;
 import org.springframework.cloud.deployer.spi.core.AppDeploymentRequest;
 import org.springframework.cloud.deployer.spi.task.LaunchState;
-import org.springframework.cloud.deployer.spi.task.TaskLauncher;
-import org.springframework.cloud.deployer.spi.test.AbstractTaskLauncherIntegrationJUnit5Tests;
-import org.springframework.cloud.deployer.spi.test.Timeout;
 import org.springframework.core.io.Resource;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.entry;
-import static org.awaitility.Awaitility.await;
 
 /**
  * Integration tests for {@link KubernetesTaskLauncher}.
  *
  * @author Thomas Risberg
  * @author Chris Schaefer
+ * @author Chris Bono
  */
 @SpringBootTest(classes = {KubernetesAutoConfiguration.class}, properties = {
 		"spring.cloud.deployer.kubernetes.namespace=default"
 })
-public class KubernetesTaskLauncherIntegrationIT extends AbstractTaskLauncherIntegrationJUnit5Tests {
-
-	@Autowired
-	private TaskLauncher taskLauncher;
-
-	@Autowired
-	private KubernetesClient kubernetesClient;
-
-	@Override
-	protected TaskLauncher provideTaskLauncher() {
-		return taskLauncher;
-	}
+public class KubernetesTaskLauncherIntegrationIT extends AbstractKubernetesTaskLauncherIntegrationTests {
 
 	@Test
-	@Override
-	@Disabled("Currently reported as failed instead of cancelled")
-	public void testSimpleCancel() throws InterruptedException {
-		super.testSimpleCancel();
-	}
-
-	@Override
-	protected String randomName() {
-		return "task-" + UUID.randomUUID().toString().substring(0, 18);
-	}
-
-	@Override
-	protected Resource testApplication() {
-		return new DockerResource("springcloud/spring-cloud-deployer-spi-test-app:latest");
-	}
-
-	@Override
-	protected Timeout deploymentTimeout() {
-		return new Timeout(20, 5000);
+	void taskLaunchedWithJobPodAnnotations(TestInfo testInfo) {
+		logTestInfo(testInfo);
+		launchTaskPodAndValidateCreatedPodWithCleanup(
```

Comment:
One thing I noticed about both the job and pod ITs is that they all did the following:
1. setup an app def/request 
2. launch the task
3. wait for the task to be running
4. verify the condition(s) under test
5. destroy the task 

I also noticed only steps 1 and 4 changed on each test. And for "step 1" it was only a different set of deployment props. 

To pull those steps out and only write them once I introduced this `launchTaskPodAndValidateCreatedPodWithCleanup` that takes in set of props (uses those when doing step 1 above) and then takes in a consumer of the resulting pod which is responsible for the assertions on the pod. 

I think this makes the test more concise and hopefully easier to add tests in the future.